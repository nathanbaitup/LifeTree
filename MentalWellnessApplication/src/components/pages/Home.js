import React, { Component } from 'react';
import { Alert, Button, Text, View, Image, Dimensions, TouchableOpacity, ImageBackground, StyleSheet } from 'react-native';

export default class Home extends Component {

    state = {
        title: 'Good Day, User!', // Added to the state as it can be updated later.
        daysUsed: 1,
        dailyStreak: 0,
        dailyStreakText: 'ðŸ”¥: ',
    }

    showDailyStreakText = () => {
        this.setState({ dailyStreakText: 'Daily Streak: ' }, () => {
            setTimeout(() => {
                this.setState({ dailyStreakText: 'ðŸ”¥: ' });
            }, 2000);
        });
    };

    render() {
        return (
            <ImageBackground source={require('../resources/img/background.png')} style = {{width:'100%', height:'100%', opacity:50}} >
            <View style={styles.mainContainer}>
                <View style={styles.heading} >
                    <Text style={styles.title}> {this.state.title} </Text>
                    <TouchableOpacity >
                        <Image style={styles.profilePic} source={{ uri: 'https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png' }} />
                    </TouchableOpacity>
                </View>
                <View onPress={this.showDailyStreakText}>
                    <TouchableOpacity style={styles.dailyStreak} onPress={this.showDailyStreakText}>
                        <Text style={styles.dailyStreakCounter} onPress={this.showDailyStreakText}> {this.state.dailyStreakText} {this.state.dailyStreak}</Text>
                    </TouchableOpacity>
                </View>
                <View style={styles.treeFrame}>
                    <Image source={require('../resources/img/trees/standard/tree-0.png')} style={styles.tree} />
                    {/* Not needed for MVP, placeholder template for Quotes API. */}
                    <Text style={styles.inspireQuote}>Anything is possible to those who believe. {'\n'} Mark: 9:23</Text>

                    <Text style={styles.daysUsed}> Days Used: {this.state.daysUsed}</Text>
                    <Button style={styles.detailsBTN}
                        onPress={showDailyUseDetails}
                        title="Find Out More"
                        accessibilityLabel='Find out more about how many used days affects the application' />
                </View>
            </View>
            </ImageBackground>
        );
    }
}

// Function used to store height of device being used for responsive design on the homescreen.
const height = Dimensions.get('window').height;
const width = Dimensions.get('window').width;

// Function that creates an alert to explain why the application should be used daily.
const showDailyUseDetails = () => {
    Alert.alert(
        // The alert title
        'Why use this app daily?',
        // The alert message
        'By using the application daily, you increase your days used counter! This has a direct link to the growth of the tree, which grows whilst you grow. ' +
        'Watch the tree grow overtime and see how far you have come on your own journey.',
        // brackets are required or android will giv an error message.
        [
            {
                text: 'OK', onPress: () => { }
            }
        ]);
};

// The styling for home page.
const styles = StyleSheet.create({
    mainContainer: {
        flexDirection: 'column',
        justifyContent: 'center',
    },
    background:{
        width: '100%',
        height: '100%',
        justifyContent: 'center',
    },
    heading: {
        flexDirection: 'row',
        paddingTop: 4,
    },
    title: {
        paddingTop: 5,
        paddingLeft: 5,
        textAlign: 'left',
        fontSize: 30,
        fontWeight: 'bold',
        alignItems: 'flex-start',
        paddingRight: width / 4
    },
    profilePic: {
        //justifyContent: 'flex-end',
        //alignItems: 'flex-end',
        width: 40,
        height: 40,
        borderRadius: 40,
    },
    tree: {
        alignSelf: 'center',
        width: 200,
        height: 125,
    },
    treeFrame: {
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        paddingTop: (height / 4)
    },
    inspireQuote: {
        fontSize: 14,
        fontStyle: 'italic',
        textAlign: 'center',
        paddingTop: 10,
        color: '#7a7a7a',
    },
    daysUsed: {
        fontSize: 18,
        fontStyle: 'italic',
        textAlign: 'center',
        paddingTop: 15,
    },
    detailsBTN: {
        paddingTop: 5,
        textTransform: 'lowercase',
    },
    dailyStreakCounter: {
        backgroundColor: '#00e676',
        borderRadius: 12,
        overflow: 'hidden',
        padding: 10,

    },
    dailyStreak: {
        flex: 1,
        flexWrap: 'wrap',
        flexDirection: 'row',
        paddingTop: 5,
        paddingLeft: 10,
    }



});
