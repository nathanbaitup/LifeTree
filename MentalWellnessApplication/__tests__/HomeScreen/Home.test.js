// https://keyholesoftware.com/2020/11/02/using-jest-and-testing-library-with-react-native-part-1-setting-up/
// helped with setting up testing.

import 'react-native';
import React from 'react';
import Home from '../../src/components/pages/HomeScreen/Home';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('Checks to see if the Home page renders', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});