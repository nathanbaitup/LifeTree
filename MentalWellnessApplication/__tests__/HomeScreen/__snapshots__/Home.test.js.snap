// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing that home component is rendered and can press on settings, learn more and daily streak Checks to see if the Home page renders 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "opacity": 50,
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/resources/img/background.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    statusBarTranslucent={true}
    supportedOrientations={
      Array [
        "landscape",
        "portrait",
      ]
    }
    transparent={true}
    visible={true}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "flex": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.25)",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <ActivityIndicator
          color="white"
          size="large"
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {},
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "bottom": 0,
                "flex": 1,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "height": 50,
                  "top": 80,
                },
                Object {
                  "color": "#FFFFFF",
                },
              ]
            }
          >
            Loading...
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`Testing that home component is rendered and can press on settings, learn more and daily streak Checks to see if the Home page renders in loading state 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "opacity": 50,
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/resources/img/background.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    statusBarTranslucent={true}
    supportedOrientations={
      Array [
        "landscape",
        "portrait",
      ]
    }
    transparent={true}
    visible={true}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "flex": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.25)",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <ActivityIndicator
          color="white"
          size="large"
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {},
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "bottom": 0,
                "flex": 1,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "height": 50,
                  "top": 80,
                },
                Object {
                  "color": "#FFFFFF",
                },
              ]
            }
          >
            Loading...
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`Testing that home component is rendered and can press on settings, learn more and daily streak Checks to see if the Home page renders with default user 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "opacity": 50,
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/resources/img/background.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    statusBarTranslucent={true}
    supportedOrientations={
      Array [
        "landscape",
        "portrait",
      ]
    }
    transparent={true}
    visible={true}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "flex": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.25)",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <ActivityIndicator
          color="white"
          size="large"
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {},
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "bottom": 0,
                "flex": 1,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "height": 50,
                  "top": 80,
                },
                Object {
                  "color": "#FFFFFF",
                },
              ]
            }
          >
            Loading...
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
